code/modules/events/event.dm

```diff
@@ -51,6 +56,9 @@
 	var/endedAt				= 0 //When this event ended.
 	var/processing_active 	= TRUE
 	var/datum/event_meta/event_meta = null
+	var/list/affecting_z	= null // List of z-levels to affect, null lets the event choose (usally station_levels)
+	var/has_skybox_image	= FALSE // True if SSskybox should query this event for an image to put in the skybox.
+	var/obj/effect/overmap/visitable/ship/victim = null // Ship that triggered this event on itself.  Some messages might be different etc.

```

Comment:
The overmap event system doesn't involve stationary sectors.  Events are triggered when a ship enters an overmap *turf*, and end when the ship leaves that turf*  A reference to the turf would not be particularly useful to the events, whereas a reference to the ship is quite useful in that it gives events access to the name of the vessel, and to the z-levels involved  (or in the case of meteors, easy access to how fast the ship is traveling etc).  If an event did want to know the turf for some reason its as simple as  `get_turf(victim)` whereas doing the opposite is much more difficult. 

I'm not a big fan of the variable naming tho, I think it would be simpler and clearer if it was simply named `ship`!

*Unless the next turf has the same event type in it, in which case the event continues until they clear that turf too, and so on.